File: rag-backend/rag/src/utils/response.py
from fastapi import Request
from fastapi.responses import JSONResponse
from lib.pydentic_models import ErrorResponse


class APIError(Exception):
    def __init__(self, message: str, status_code: int = 500, details: any = None):
        self.message = message
        self.status_code = status_code
        self.details = details
        super().__init__(self.message)


async def api_error_handler(request: Request, exc: APIError):
    error = ErrorResponse(
        status="error", message=exc.message, code=exc.status_code, details=exc.details
    )
    return JSONResponse(status_code=exc.status_code, content=error.model_dump())

File: rag-backend/api/src/config/response.ts
export type SuccessResponse<T> = {
	status: "success" | string;
	message: string;
	data: T;
};
export function success<T>(
	data: T,
	message: string,
	status: string,
): SuccessResponse<T> {
	return {
		status: status || "success",
		message,
		data,
	};
}
export type ErrorResponse = {
	success: false;
	error: {
		message: string;
		code?: string;
	};
};


export function error(message: string, code?: string): ErrorResponse {
	return {
		success: false,
		error: { message, code },
	};
}